src/rgw/rgw_rados.cc

```diff
@@ -11967,6 +11961,73 @@ int RGWRados::get_bucket_info(RGWObjectCtx& obj_ctx,
   return 0;
 }
 
+int RGWRados::get_bucket_info(RGWObjectCtx& obj_ctx,
+                              const string& tenant, const string& bucket_name,
+                              RGWBucketInfo& info,
+                              real_time *pmtime, map<string, bufferlist> *pattrs)
+{
+  bucket_info_entry e;
+  string bucket_entry;
+  rgw_make_bucket_entry_name(tenant, bucket_name, bucket_entry);
+
+  if (binfo_cache->find(bucket_entry, &e)) {
+    info = e.info;
+    if (pattrs)
+      *pattrs = e.attrs;
+    if (pmtime)
+      *pmtime = e.mtime;
+    return 0;
+  }
+
+  return bucket_info_cache_miss(obj_ctx, tenant, bucket_name, info, pmtime,
+                                pattrs);
+}
+
+bool RGWRados::try_refresh_bucket_info(RGWBucketInfo& info,
+                                       ceph::real_time *pmtime,
+                                       map<string, bufferlist> *pattrs)
+{
+  RGWObjectCtx obj_ctx(this);
+  bucket_info_entry e;
+  string bucket_entry;
+  rgw_make_bucket_entry_name(info.bucket.tenant, info.bucket.name, bucket_entry);
+
+  auto checkv = info.objv_tracker.read_version;
```

Comment:
Note that there's a string copy here so it'd be better to take a reference. Maybe use reference here, and if this check fails only then assign it to a local variable?